src/LiveComponent/assets/src/get_array_value.ts

```
@@ -0,0 +1,39 @@
+/**
+ * Resolve multiple value data from changed HTML element.
+ *
+ * @param element       Current HTML element
+ * @param value         Resolved value of a single HTML element (.value or [data-value])
+ * @param currentValue  Current data value
+ */
+export function getArrayValue(
+    element: HTMLElement,
+    value: string|null,
+    currentValue: any
+): Array<string> {
+    // Enforce returned value is an array
+    if (!(currentValue instanceof Array)) {
+        currentValue = [];
+    }
+
+    if (element instanceof HTMLInputElement && element.type === 'checkbox') {
+        const index = currentValue.indexOf(value);
+
+        if (element.checked) {
+            // Add value to an array if it's not in it already
+            if (index === -1) {
+                currentValue.push(value);
+            }
+        } else {
+            // Remove value from an array
+            if (index > -1) {
+                currentValue.splice(index, 1);
+            }
+        }
+    } else if (element instanceof HTMLSelectElement) {
+        // Select elements with `multiple` option require mapping chosen options to their values
+        currentValue = Array.from(element.selectedOptions).map(el => el.value);
+    }
```

Is there an "else" case? Is there a case where we hit this function but the element is NOT a checkbox or an `HTMLSelectElement`? 

If there "shouldn't" be any other case, then we should throw an exception. That way, if there IS a case, it can be reported to us.

---

This function is called whenever array data is expected (defined by `[]` at the end of model name), so user can write broken code that will trigger it with incorrect element passed. Therefore I'll go with a `throw` in the final step :+1: